### PR TITLE
sys/net/netutils: add netutils_get_ipv4()

### PR DESCRIPTION
**Pull request by HendrikVE**
### Contribution description

Add function `netutils_get_ipv4()` to `netutils`.

### Testing procedure

Existing test in `tests/netutils` was extended.

Can be tested with #17763

### CODE REVIEW

**Comment by HendrikVE** [comment]
> After https://github.com/RIOT-OS/RIOT/pull/16688 was merged I tried to port the examples to used URIs (utilizing the uri_parser) module, which was halted, since netutils_get_ipv6() requires a string (\0 terminated char array) and not a char buffer (a char array + length), the latter being the output of the non-destructive uri_parser. Maybe in the design of netutils_get_ipv4() this lack in the original netutils_get_ipv6() can already be addressed?

Unfortunately it can't, because the root of the problem is that `sock_dns_query`, which is used by `netutils_get_ipv4` and `netutils_get_ipv6`, requires a null-terminated string .

**Comment by miri64** [comment]
> Unfortunately it can't, because the root of the problem is that `sock_dns_query`, which is used by `netutils_get_ipv4` and `netutils_get_ipv6`, requires a null-terminated string .

That sounds also like a fixable problem ;-), but I can understand if you are not willing to do it here. :-)

**Comment by HendrikVE** [comment]
Yeah I think that's something for another PR, but I can add it to my list ;)

File: sys/include/net/utils.h [hunk 1]
```diff
@@ -24,13 +25,27 @@
...
 extern "C" {
 #endif
 
+/**
+ * @brief   Parse an IPv4 address / hostname string.
+ *          If the @ref net_sock_dns module is used, this will
+ *          attempt to resolve hostnames via DNS to IPv4 addresses.
+ *
+ * @param[out]  addr        IPv4 address of the host
+ * @param[out]  netif       Interface if address is link-local
```
Comment (benpicco) [inline sys/include/net/utils.h:42]:
```suggestion
```

File: sys/net/netutils/util.c [hunk 2]
```diff
@@ -33,6 +35,39 @@ static bool _netif_get(netif_t **current_netif)
...
+{
+    if (hostname == NULL) {
+        return -EINVAL;
+    }
+
+    for (size_t i = 0; i < strlen(hostname); i++) {
+        bool is_not_ipv4 = (hostname[i] < '0' || hostname[i] > '9') && hostname[i] != '.';
+
+#ifdef MODULE_SOCK_DNS
+        /* once we see an invalid character for an IPv4 address try to resolve the hostname by DNS */
```
Comment (benpicco) [inline sys/net/netutils/util.c:48]:
```suggestion
        /* once we see an invalid character for an IPv4 address 
         * try to resolve the hostname by DNS
         */
```

File: sys/net/netutils/util.c [hunk 3]
```diff
@@ -42,8 +77,7 @@ int netutils_get_ipv6(ipv6_addr_t *addr, netif_t **netif, const char *hostname)
     }
 
 #ifdef MODULE_SOCK_DNS
-    /* hostname is not an IPv6 address */
-    if (strchr(hostname, ':') == NULL) {
+    if (strchr(hostname, ':') == NULL) { /* hostname is not an IPv6 address */
```
Comment (benpicco) [inline sys/net/netutils/util.c:80]: Why can't this stay where it was?

**Review by benpicco — COMMENTED** [review]
Looks good to me!
Just some small nitpicks:

File: sys/include/net/utils.h (hunk 1 above)
Comment (HendrikVE) [reply sys/include/net/utils.h]: Removed

File: sys/net/netutils/util.c (hunk 2 above)
Comment (HendrikVE) [reply sys/net/netutils/util.c]: Shortened

File: sys/net/netutils/util.c (hunk 3 above)
Comment (HendrikVE) [reply sys/net/netutils/util.c]: Force of habit, sorry^^ I tend to put the description for an expression next to it. I've put it back.

**Review by benpicco — APPROVED** [review]

**Pull request merged**